[DPG] Incorrect next page function name generated in Metrics Advisor

For [listMetricFeedbacks](https://github.com/Azure/azure-rest-api-specs/blob/08f5e391f2153a99580b458cc71ef88e45dd0531/specification/cognitiveservices/data-plane/MetricsAdvisor/preview/v1.0/MetricsAdvisor.json#L1574) method, next link function is generated as `GetMetricFeedbacksNexts` instead of `GetMetricFeedbacksNext`. [Here](https://github.com/ShivangiReja/azure-sdk-for-net/blob/MetricsAdvisor-Experiment/sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Generated/MetricsAdvisorClient.cs#L2304) is the generated method.

Pipeline is failing with the below error:

>CSpell: Files checked: 2, Issues found: 2 in 1 files
cspell run complete, restoring original configuration and removing temp file.
./sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs:211:76 - **Unknown word (Nexts)**
./sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs:212:81 - **Unknown word (Nexts)**
##[error]Spelling errors detected. To correct false positives or learn about spell checking see: https://aka.ms/azsdk/engsys/spellcheck

Check [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1438976&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=d34cfe88-f3e1-5ca1-ec50-18a5cea1099a&l=87) for more details.

--- a/src/AutoRest.CSharp/Common/Utilities/StringExtensions.cs
+++ b/src/AutoRest.CSharp/Common/Utilities/StringExtensions.cs
@@ -411,6 +411,11 @@ namespace AutoRest.CSharp.Utilities
                     newName = pluralMethodName;
                 }
                 newName = newName.ReplaceFirst("List", getMethodPrefix);
+                // For the next page method of List operations, Next is appended to the end of a List method and ListXXXNext will be renamed to GetXXXNexts. Here we need to check and fix this case by removing the "s" in the end.
+                if (words.Last().Equals("Next", StringComparison.Ordinal))
+                {
+                    newName = newName.TrimEnd('s');
+                }
             }
             return newName;
         }

--- a/test/AutoRest.TestServer.Tests/Common/StringExtensionsTests.cs
+++ b/test/AutoRest.TestServer.Tests/Common/StringExtensionsTests.cs
@@ -170,6 +170,7 @@ namespace AutoRest.CSharp.Utilities
         [TestCase("ListByFarmerId", "ApplicationData", "GetAllApplicationDataByFarmerId")]
         [TestCase("ListDataTip", "", "GetDataTips")]
         [TestCase("ListTipData", "", "GetAllTipData")]
+        [TestCase("ListMetricFeedbacksNext", "", "GetMetricFeedbacksNext")]
         public void ValidateRenameListToGet(string methodName, string resourceName, string expected)
         {
             var result = methodName.RenameListToGet(resourceName);

